python/src/spark_rapids_ml/umap.py

```diff
@@ -1441,19 +1422,48 @@ def _out_schema(self, input_schema: StructType) -> Union[StructType, str]:
         return f"array<{pyspark_type}>"
 
     def write(self) -> MLWriter:
-        return _CumlModelWriterNumpy(self)
+        return _CumlModelWriterParquet(self)
 
     @classmethod
     def read(cls) -> MLReader:
-        return _CumlModelReaderNumpy(cls)
+        return _CumlModelReaderParquet(cls)
 
 
-class _CumlModelWriterNumpy(_CumlModelWriter):
+class _CumlModelWriterParquet(_CumlModelWriter):
     """
-    Override parent writer to save numpy objects of _CumlModel to the file
+    Override parent writer to write _CumlModel array attributes to Spark Dataframe as parquet
     """
 
     def saveImpl(self, path: str) -> None:
+
+        spark = _get_spark_session()
+
+        def write_sparse_array(array: scipy.sparse.spmatrix, df_path: str) -> None:
+            schema = StructType(
+                [
+                    StructField("indices", ArrayType(IntegerType(), False), False),
+                    StructField("indptr", ArrayType(IntegerType(), False), False),
+                    StructField("data", ArrayType(FloatType(), False), False),
+                    StructField("shape", ArrayType(IntegerType(), False), False),
+                ]
+            )
+            data_df = spark.createDataFrame(
```

Comment:
By putting all data in a single row, what are the limits on data sizes?  Here and below for dense?   Should add these limits to documentation.

Alternative is to create regular possibly multiple dataframes, save in multiple files/subfolders and add a column to order/sort by to restore original order.